File: src/getml-app/src/tcp/logger.go
// Copyright 2025 Code17 GmbH
// 
// This file is licensed under the Elastic License 2.0 (ELv2). 
// Refer to the LICENSE.txt file in the root of the repository 
// for details.
// 

package tcp

import (
	"net"
)

type logger struct {
	c net.Conn
}

func newLogger(c net.Conn) *logger {
	return &logger{c: c}
}

func (l *logger) log(msg string) {
	sendString(l.c, "log: "+msg)
}

File: src/engine/src/communication/Logger.cpp
// Copyright 2022 The SQLNet Company GmbH
//
// This file is licensed under the Elastic License 2.0 (ELv2).
// Refer to the LICENSE.txt file in the root of the repository
// for details.
//

#include "communication/Logger.hpp"

#include "communication/Sender.hpp"

namespace communication {

void Logger::log(const std::string& _msg) const {
  assert_true(monitor_);

  const auto now = std::chrono::system_clock::now();

  const std::time_t current_time = std::chrono::system_clock::to_time_t(now);

  std::cout << std::ctime(&current_time) << _msg << std::endl << std::endl;
}

}  // namespace communication
